### PR TITLE
Updated Flink

### PR DESCRIPTION
**Pull request by dawidsula26**

### CODE REVIEW

File: flink114/model/src/test/scala/pl/touk/nussknacker/compatibility/MockKafkaComponentProvider.scala [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package pl.touk.nussknacker.compatibility
+
+import pl.touk.nussknacker.compatibility.common.MockSchemaRegistry.schemaRegistryMockClient
+import pl.touk.nussknacker.engine.flink.util.transformer.FlinkKafkaComponentProvider
+import pl.touk.nussknacker.engine.schemedkafka.schemaregistry.confluent.client.{ConfluentSchemaRegistryClientFactory, MockConfluentSchemaRegistryClientFactory}
+
+class MockKafkaComponentProvider extends FlinkKafkaComponentProvider {
```
Comment (mproch) [inline flink114/model/src/test/scala/pl/touk/nussknacker/compatibility/MockKafkaComponentProvider.scala]: I wonder, maybe let's move it to common?

File: build.sbt [hunk 2]
```diff
@@ -118,6 +120,65 @@ lazy val flink111ManagerCompat = (project in file("flink111/manager")).
...
+val flinkExclusionsForBefore1_15 = Seq(
+  excludeDependencies ++= List(
+    "org.apache.flink" % "flink-streaming-java",
+    "org.apache.flink" % "flink-statebackend-rocksdb",
+    "org.apache.flink" % "flink-connector-kafka",
+    "org.apache.flink" % "flink-test-utils"
+  )
+)
+
+def flinkDependenciesCommonForBefore1_15(version: String) = Seq(
```
Comment (mproch) [inline build.sbt]: I think you have to figure provided/test here

File: build.sbt [hunk 3]
```diff
@@ -118,6 +121,78 @@ lazy val flink111ManagerCompat = (project in file("flink111/manager")).
...
 
+
+lazy val flink114ModelCompat = (project in file("flink114/model")).
+  settings(commonSettings(scala212V)).
+  settings(flinkSettingsCommonForBefore1_15(flink114V)).
+  settings(
+    name := "flink114-model",
+    libraryDependencies ++= deps(flink114V),
+    dependencyOverrides ++= Seq(
+      "org.apache.kafka" % "kafka-clients" % kafkaV
```
Comment (mproch) [inline build.sbt:138]: maybe we can add it to commons? It's used every time?

File: flink114/manager/src/main/scala/pl/touk/nussknacker/engine/management/CustomFlinkStreamingDeploymentManagerProvider.scala [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package pl.touk.nussknacker.engine.management
+
+import pl.touk.nussknacker.engine.management.common.CommonFlinkStreamingDeploymentManagerProvider
```
Comment (mproch) [inline flink114/manager/src/main/scala/pl/touk/nussknacker/engine/management/CustomFlinkStreamingDeploymentManagerProvider.scala]: Why do we need it at all?

Comment (dawidsula26) [reply flink114/manager/src/main/scala/pl/touk/nussknacker/engine/management/CustomFlinkStreamingDeploymentManagerProvider.scala]: We need to return custom StreamingRestManager that differently handles checking if job is running. I'm not sure why it is necessary, but it was there before and timeouts happen without it.

File: build.sbt (hunk 3 above)
Comment (dawidsula26) [reply build.sbt:138]: It was not necessary in managers. I removed it from there

**Review by mproch — APPROVED** [review]
:+1: that was hard one... ;)

**Pull request merged**